driver/src/main/java/org/neo4j/driver/internal/security/TLSSocketChannel.java

```diff
@@ -81,7 +82,7 @@ public static TLSSocketChannel create( ByteChannel channel, Logger logger, SSLEn
         }
         catch ( SSLHandshakeException e )
         {
-            throw new ClientException( "Failed to establish secured connection with the server: " + e.getMessage(), e );
+            throw new UnauthorizedException( "Failed to establish secured connection with the server: " + e.getMessage(), e );
```

Comment:
I'm not sure that we should reuse `UnauthorizedException` for TLS handshake failure as well as for actual login failure. The Python driver uses `ServiceUnavailable` here but that doesn't seem quite right either.

Perhaps we should have something separate such as `SecurityLayerUnavailable`?